schema-bridge.graphql

```diff
@@ -428,15 +897,338 @@ type PoolHourlySnapshot @entity {
+
+  " The equivalent of our token in the other chain "
+  crosschainToken: CrosschainToken!
+
+  " Indicates is the transfer involved a swap while bridging "
+  isSwap: Boolean!
+
+  " The transaction ID on the other chain, if available "
+  crossTransactionID: Bytes
+
+  " Block number of this event "
+  blockNumber: BigInt!
+
+  " Timestamp of this event "
+  timestamp: BigInt!
+}
+
+# Some bridges support sending
+type BridgeMessage implements Event @entity(immutable: true) {
+  " event.transaction.hash.concatI32(event.logIndex) "
+  id: Bytes!
+
+  " Transaction hash of the transaction that emitted this event "
+  hash: Bytes!
+
+  " Event log index. For transactions that don't emit event, create arbitrary index starting from 0 "
+  logIndex: Int!
+
+  " The protocol this transaction belongs to "
+  protocol: BridgeProtocol!
+
+  " Account that sent or received the message. Should match with either transferFrom or transferTo "
+  account: Account!
+
+  " Address that received the tokens "
+  to: Bytes!
+
+  " Address that sent the tokens "
+  from: Bytes!
+
+  " Is it an ougoing message to some other chain or incoming instead? "
+  isOutgoing: Boolean!
+
+  " Chain ID from where the message is being sent "
+  fromChainID: Int!
+
+  " Chain ID where the message is being sent to "
+  toChainID: Int!
+
+  " Address sending the message. Here if it is outgoing, or on the other chain if incoming "
+  transferFrom: Bytes
+
+  " Address receiving the message. Here if it is incoming, or on the other chain if outgoing "
+  transferTo: Bytes
+
+  " Contents of the message "
+  data: Bytes!
+
+  " The transaction ID on the other chain, if available "
+  crossTransactionID: Bytes
+
+  " Block number of this event "
+  blockNumber: BigInt!
+
+  " Timestamp of this event "
+  timestamp: BigInt!
+}
+
 # An account is a unique Ethereum address
 # Helps to accumulate total unique users
 type Account @entity {
   " Address of the account "
-  id: ID!
+  id: Bytes!
+
+  " ChainIDs where this account has sent or received funds through this bridge "
+  chains: [Int!]!
+
+  " Total number of transactions this account has sent through this bridge originating from this chain "
+  transferOutCount: Int!
+
+  " Total number of transactions this account has received through this bridge originating on other chains "
+  transferInCount: Int!
+
+  " All transfer events for this account "
+  transfers: [BridgeTransfer!]! @derivedFrom(field: "account")
+
+  " Total number of liquidity deposits this account has made "
+  depositCount: Int!
+
+  " All deposit events for this account "
+  deposits: [LiquidityDeposit!]! @derivedFrom(field: "account")
+
+  " Total number of liquidity withdrawals this account has made "
+  withdrawCount: Int!
+
+  " All withdraw events for this account "
+  withdraws: [LiquidityWithdraw!]! @derivedFrom(field: "account")
+
+  " Total number of messages this account has sent or received through this bridge "
+  messageCount: Int!
```

Comment:
suggest to keep `messageSentCount` (optionally also `messageReceivedCount`) instead, since this field will be used to calculate `protocol.cumulativeUniqueMessageSenders`